op_crates/web/lib.deno_web.d.ts

```diff
@@ -178,32 +178,84 @@ declare function atob(s: string): string;
  */
 declare function btoa(s: string): string;
 
-declare class TextDecoder {
-  /** Returns encoding's name, lowercased. */
+/** A decoder for a specific method, that is a specific character encoding, like utf-8, iso-8859-2, koi8, cp1261, gbk, etc. A decoder takes a stream of bytes as input and emits a stream of code points. For a more scalable, non-native library, see StringView – a C-like representation of strings based on typed arrays. */
+interface TextDecoder extends TextDecoderCommon {
+  /**
+     * Returns the result of running encoding's decoder. The method can be invoked zero or more times with options's stream set to true, and then once without options's stream (or set to false), to process a fragmented input. If the invocation without options's stream (or set to false) has no input, it's clearest to omit both arguments.
+     *
+     * ```
+     * var string = "", decoder = new TextDecoder(encoding), buffer;
+     * while(buffer = next_chunk()) {
+     *   string += decoder.decode(buffer, {stream:true});
+     * }
+     * string += decoder.decode(); // end-of-queue
+     * ```
+     *
+     * If the error mode is "fatal" and encoding's decoder returns error, throws a TypeError.
+     */
+  decode(input?: BufferSource, options?: TextDecodeOptions): string;
+}
+
+declare var TextDecoder: {
+  prototype: TextDecoder;
+  new (label?: string, options?: TextDecoderOptions): TextDecoder;
+};
+
+interface TextDecoderCommon {
+  /**
+     * Returns encoding's name, lowercased.
+     */
   readonly encoding: string;
-  /** Returns `true` if error mode is "fatal", and `false` otherwise. */
+  /**
+     * Returns true if error mode is "fatal", otherwise false.
+     */
   readonly fatal: boolean;
-  /** Returns `true` if ignore BOM flag is set, and `false` otherwise. */
-  readonly ignoreBOM = false;
-  constructor(
-    label?: string,
-    options?: { fatal?: boolean; ignoreBOM?: boolean },
-  );
-  /** Returns the result of running encoding's decoder. */
-  decode(input?: BufferSource, options?: { stream?: false }): string;
-  readonly [Symbol.toStringTag]: string;
+  /**
+     * Returns the value of ignore BOM.
+     */
+  readonly ignoreBOM: boolean;
+}
+
+interface TextDecodeOptions {
+  // TODO(caspervonb) support streaming.
```

Comment:
This doesn't make sense in the type definitions, it should be located in the runtime code.

Reply:
The old ad-hoc type definitions only allow `undefined | false` so kept that quirk because it was already there.
Am fine with removing it tho.